File: python/first.py
import sys
import time
from itertools import islice

start = time.time()
words = {}
with open(sys.argv[1]) as file:
  for word in file.read().split():
    word = word.lower()
    words[word] = words.setdefault(word, 0) + 1

top_length = 10
if len(sys.argv) == 3:
  top_length = int(sys.argv[2])
top_words = list(islice(words.items(), top_length))
for word, n in words.items():
  for i, top in enumerate(top_words):
    if n > top[1]:
      tmp = top_words[i:-1]
      top_words[i] = (word, n)
      top_words = top_words[0:i+1] + tmp
      break

print(top_words)
end = time.time()
print('elapsed: ' + str((end-start)*1000) + 'ms')

File: node/first.js
import { readFileSync } from 'fs';

console.time('elapsed')
const words = {};
const fileName = 'alice_in_wonderland.txt'

const file = readFileSync(fileName, 'utf-8')
file.split(/(\s+)/).filter((e) => e.trim().length > 0).forEach(word => {
  word = word.toLowerCase()
  words[word] = 1 + (words[word] || 0)
})

let topWords = Array.from(Object.entries(words)).slice(0, 10)
Object.entries(words).forEach(([word, n]) => {
  for (const [i, top] of topWords.entries()) {
    if (n > top[1]) {
      const tmp = topWords.slice(i, topWords.length-1)
      topWords[i] = [word, n]
      topWords = topWords.slice(0, i+1).concat(tmp)
      break
    }
  }
})

console.log(topWords)
console.timeEnd('elapsed')
